### ISSUE TITLE
mfsmount suddenly drops away

### ISSUE THREAD

**Issue by matthiaz** [lizardfs/lizardfs#576]
Description: Ubuntu 16.04.2 LTS
Architecture: amd64
Source: lizardfs
Version: 3.11.2

I have several mounts mounted using 
mfsmount /mnt/lizardfs

This works fine, but sometimes the connection drops away. In syslog I see this mentioned:
Jul 12 13:03:20 PHP2 kernel: [6584884.172457] mfsmount[9147]: segfault at 0 ip 0000000000453809 sp 00007fb88bffeaa0 error 4 in mfsmount[400000+c5000]

on mfsmaster I see:
Jul 12 13:03:20 lizardmaster mfsmaster[21168]: connection with client(ip:172.16.16.112) has been closed by peer

To fix it, I have to do:
umount /mnt/lizardfs && mfsmount /mnt/lizardfs

Any idea on why the mfsmount crashes? And if not, is there a way that I can set up the system so it automatically reconnects the mount when this happens?

**Comment by psarna** [comment]
What's your LizardFS release and operating system?

**Comment by matthiaz** [comment]
Updated my post with version info.

**Comment by matthiaz** [comment]
Any update on this? I had the same problem again this morning.
Jul 13 08:58:32 PHP2 kernel: [6656652.591504] mfsmount[4700]: segfault at 59667fe6 ip 0000000000453809 sp 00007f5d669faaa0 error 4 in mfsmount[400000+c5000]

**Comment by psarna** [comment]
I cannot easily reproduce it. Do you have any idea if any specific operations cause the mount to fail? Debugging information shows that the segfault happened inside a symlink() call, which is not very likely.

Also, can you provide output of commands below?
~~~
md5sum $(which mfsmount)
ldd $(which mfsmount)
mfsmount --version
~~~

**Comment by matthiaz** [comment]
Could you point me to the line in the code where the symlink happens ? 
 
root@PHP2:/mnt/lizardfs# **md5sum $(which mfsmount)**
80568e5e411f069a324d4ed0fa88ef9f  /usr/bin/mfsmount
root@PHP2:/mnt/lizardfs# **ldd $(which mfsmount)**
        linux-vdso.so.1 =>  (0x00007ffec38de000)
        libfuse.so.2 => /lib/x86_64-linux-gnu/libfuse.so.2 (0x00007fdc1f217000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fdc1ee95000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fdc1eb8e000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fdc1e978000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fdc1e75a000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fdc1e394000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fdc1e190000)
        /lib64/ld-linux-x86-64.so.2 (0x000056147ad50000)
root@PHP2:/mnt/lizardfs# **mfsmount --version**
LizardFS version 3.11.2
FUSE library version: 2.9.2
fusermount version: 2.9.2

**Comment by psarna** [comment]
Well, now I can't, because the debug symbols are clearly mismatched.
Typically, one can extract the line of code by running the binary with debug symbols (which can be found in lizardfs-dbg package). But, since all we have here is an address from instruction pointer registry (ip 0000000000453809), it has to be exactly the same binary or debug symbols will be useless.

I just downloaded 3.11.2 client package straight from http://packages.lizardfs.com/ubuntu/xenial/pool/main/l/lizardfs/ and it shows different md5sum output:
~~~
md5sum ./mfsmount
262080c5aa7bebaa4d4b719e43b092e9  ./mfsmount
~~~
Your mfsmount md5sum, in turn, matches our ubuntu-trusty (14.04) packages, and I just realise it's because our Download page lacks updated information for Ubuntu 16...

Please use 'xenial' instead of 'trusty' in these instruction and reinstall, while I tell someone to update the page. (And, in the meantime, check with debug symbols again, this time using ubuntu 14 package, maybe it tells something).

**Comment by psarna** [comment]
Aaaand it makes perfect sense now. These are bugs and fixes are backported to 3.11.x already, but 3.11.3 is not yet released. Three top commits from this: https://github.com/lizardfs/lizardfs/commits/3.11 fix your issues. You can either wait for 3.11.3 (coming very soon) or build from source.

**Comment by pkali** [comment]
The version 3.11.3 fixes this issue, thanks @matthiaz for speeding up the release!
